packages/box/src/Box.spec.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import { render, cleanup } from '@testing-library/react';
+import Box from '.';
+import { BoxProps } from './Box';
+
+// types
+interface SharedProps {
+  name?: string;
+}
+
+interface RenderedEls {
+  box?: HTMLElement | null;
+  child?: HTMLElement | null;
+}
+
+interface LinkWrapperProps {
+  href: string;
+  target: string;
+  children: React.ReactNode;
+}
+
+type testProps = BoxProps<SharedProps>;
+type unmountQueueType = Array<() => boolean>;
+
+describe('packages/box', () => {
+  const unmountQueue: unmountQueueType = [];
+  const renderedEls: RenderedEls = {};
+  const linkWrapperFn = jest.fn();
+
+  function LinkWrapper({ href, target, children, ...rest }: LinkWrapperProps) {
+    linkWrapperFn();
+    return (
+      <span {...rest}>
+        <a data-testid="link-component" href={href} target={target}>
+          {children}
+        </a>
+      </span>
+    );
+  }
+
+  const sharedProps = { name: 'testName' };
+  const anchorProps = {
+    href: 'https://cloud.mongodb.com',
+    target: '_blank',
+    ...sharedProps,
+  };
+  const linkProps = { component: LinkWrapper, ...anchorProps };
+
+  const renderBox = (props: testProps = sharedProps) => {
+    const { queryByTestId, unmount } = render(
+      <Box data-testid="box" {...props}>
+        <div data-testid="child">Child Content</div>
+      </Box>,
+    );
+
+    renderedEls.child = queryByTestId('child');
+    renderedEls.box = queryByTestId('box');
+    unmountQueue.push(unmount);
```

Comment:
This seems un-necessarily complex. 

Are we testing anything to do with unmounting?

If not can e just remove this complexity and rely on the builtin cleanup mechanism?

Reply:
@seanmhanson you wrote these tests, so defer to you for a response

Reply:
I'd like to remove this so we don't have overly complex tests. Do the tests still pass with this removed?

Reply:
Removed it

Reply:
Don't see it. 👀

Reply:
Failed to push, I see that now 🤦‍♂